rust/tools/authorization-logic/src/main.rs

```diff
@@ -25,14 +25,38 @@ mod souffle;
 mod test;
 
 use std::env;
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+#[structopt(name = "auth-logic", about = "An auth logic compiler.")]
+struct Opt {
+    /// The name of the auth logic program to compile
+    // First positional argument (should be passed as a raw value, not --filename=...)
+    #[structopt(required = true)]
+    filename: String,
+
+    /// Input directory where the auth logic program lives.
+    /// If unspecified, it will be the current working directory.
+    // Can be passed as -i or --in-dir
+    #[structopt(short, long, env("PWD"))]
+    in_dir: String,
+
+    /// Output directory, where the compiled datalog program will be saved.
+    /// If unspecified, it will be the current working directory.
+    // Can be passed as -o or --out-dir
+    #[structopt(short, long, env("PWD"))]
+    out_dir: String,
+
+    /// List of declarations to skip when generating Souffle code
+    // Can be passed as -s or --skip. Passed as a comma-separated list.
+    #[structopt(short = "s", long = "skip")]
+    decl_skip: Vec<String>,
+}
 
 fn main() {
-    let args: Vec<String> = env::args().collect();
-    let filename = &args[1];
-    let in_dir = &args[2];
-    let out_dir = &args[3];
-    let decl_skip = if(args.len() > 4) { &args[4] } else { "" };
-    compilation_top_level::compile(filename, in_dir, out_dir, &decl_skip);
-    souffle::souffle_interface::run_souffle(&format!("{}/{}.dl", out_dir,
-                                                    filename), out_dir);
+    let opt = Opt::from_args();
+    compilation_top_level::compile(&opt.filename, &opt.in_dir, &opt.out_dir,
+        &opt.decl_skip);
```

Comment:
As long as we are messing with options, it would be nice to have an option that allowed this binary to stop after compilation and just print the datalog file rather than always running Souffle afterwards. Again, feel free to ignore if this is the wrong commit for this.

Reply:
Definitely agreed. That's pretty easy to add, so I included it here.